scipy/stats/_entropy.py

```diff
@@ -196,7 +257,60 @@ def differential_entropy(
     logs = np.log(
         len(differences) * differences / (2 * window_length),
     )
-    if base is not None:
-        logs /= np.log(base)
 
     return np.mean(logs, axis=0)
+
+
+def _pad_along_last_axis(X, m):
+    """Pad the data for computing the rolling window difference"""
+    # scales a  bit better than method in _vasicek_like_entropy
+    shape = np.array(X.shape)
+    shape[-1] = m
+    Xl = np.broadcast_to(X[..., [0]], shape)  # [0] vs 0 to maintain shape
+    Xr = np.broadcast_to(X[..., [-1]], shape)
+    return np.concatenate((Xl, X, Xr), axis=-1)
+
+
+def _van_es_entropy(X, m):
+    """Compute the van Es estimator as described in [6]"""
+    # No equation number, but referred to as HVE_mn.
+    # Typo: there should be a log within the summation.
+    n = X.shape[-1]
+    term1 = 1/(n-m) * np.sum(np.log((n+1)/m * (X[..., m:] - X[..., :-m])),
+                             axis=-1)
+    k = np.arange(m, n+1)
+    return term1 + np.sum(1/k) + np.log(m) - np.log(n+1)
+
+
+def _ebrahimi_entropy(X, m):
+    """Compute the Ebrahimi estimator as described in [6]"""
+    # No equation number, but referred to as HE_mn
+    n = X.shape[-1]
+    X = _pad_along_last_axis(X, m)
+
+    differences = X[..., 2 * m:] - X[..., : -2 * m:]
+
+    i = np.arange(1, n+1).astype(float)
+    ci = np.ones_like(i)*2
+    ci[i <= m] = 1 + (i[i <= m] - 1)/m
+    ci[i >= n - m + 1] = 1 + (n - i[i >= n-m+1])/m
+
+    logs = np.log(n * differences / (ci * m))
+    return np.mean(logs, axis=-1)
+
+
+def _correa_entropy(X, m):
+    """Compute the Correa estimator as described in [6]"""
+    # No equation number, but referred to as HC_mn
+    n = X.shape[-1]
+    X = _pad_along_last_axis(X, m)
+
+    i = np.arange(1, n+1)
+    dj = np.arange(-m, m+1)[:, None]
+    j = i + dj
+    j0 = j + m - 1  # 0-indexed version of j
+
+    Xibar = np.mean(X[..., j0], axis=-2, keepdims=True)
+    num = np.sum((X[..., j0] - Xibar)*(j-i), axis=-2)
+    den = n*np.sum((X[..., j0] - Xibar)**2, axis=-2)
+    return -np.mean(np.log(num/den), axis=-1)
```

Comment:
I can see some PEP8 whitespace corrections to be made here, but not necessarily those recommended. PEP8 recommends:
> If operators with different priorities are used, consider adding whitespace around the operators with the lowest priority(ies). Use your own judgment; however, never use more than one space, and always have the same amount of whitespace on both sides of a binary operator

So I don't think I would add the whitespace around `**`, for instance. The whitespace around single operators inside functions (e.g. `np.arange(1, n+1)` and `np.log(num / den)`) is optional, and I prefer to omit because it is the highest-priority operation (e.g. add before creating the array, divide before taking the log). On the other hand, when operators are not inside functions, I add the whitepace (e.g. `j = i + dj`) because it looks a bit better - in fact, that's given as a "correct" example in the PEP8 guide.

One thing that is more difficult to justify is my space in `X[..., j0] - Xibar`. I suppose I do that because the indexing is like a high priority operator. (For example, if it were `X**2 - Xibar`, I would want the space, even though it appears inside a function. But if we refactor as [above](https://github.com/scipy/scipy/pull/13845/files#r611782368), then it wouldn't matter anymore.

Reply:
OK sure it makes sense 👍